scib/metrics/pcr.py

```diff
@@ -216,18 +271,39 @@ def pc_regression(
     else:
         if verbose:
             print("one-hot encode categorical values")
-        covariate = pd.get_dummies(covariate)
+        covariate = pd.get_dummies(covariate).to_numpy()
 
     # fit linear model for n_comps PCs
-    r2 = []
-    for i in range(n_comps):
-        pc = X_pca[:, [i]]
-        lm = LinearRegression()
-        lm.fit(covariate, pc)
-        r2_score = np.maximum(0, lm.score(covariate, pc))
-        r2.append(r2_score)
+    if verbose:
+        print(f"Use {n_threads} threads for regression...")
+    if n_threads == 1:
+        r2 = []
+        for i in tqdm(range(n_comps), total=n_comps):
+            r2_score = linreg_method(X=covariate, y=X_pca[:, [i]])
+            r2.append(np.maximum(0, r2_score))
+    else:
+        with ThreadPoolExecutor(max_workers=n_threads) as executor:
+            run_r2 = executor.map(linreg_method, [covariate] * n_comps, X_pca.T)
```

Comment:
What is parallelized here? And why is `X_pca` transposed when there are multiple threads but not the single one? Having some comments in the code would be nice :)

Reply:
I'm parallelising over all the PC components and X_pca needs to be transposed so that the matrix columns are passed (not the rows). But good point, will add a comment